File: examples/python/minimal-order-service/tests/test_adapters.py
"""Tests for adapters."""
from domain.money import Money
from domain.order import Order
from adapters.outbound.in_memory_repository import InMemoryOrderRepository
from adapters.outbound.fake_payment_gateway import FakePaymentGateway


def test_in_memory_repository_save_and_find():
    """Test saving and retrieving an order."""
    repository = InMemoryOrderRepository()
    total = Money(10000, "USD")
    order = Order("order-1", total)
    
    repository.save(order)
    
    retrieved = repository.find_by_id("order-1")
    assert retrieved is not None
    assert retrieved.id == order.id
    assert retrieved.total == order.total


def test_in_memory_repository_find_non_existent():
    """Test finding a non-existent order."""
    repository = InMemoryOrderRepository()
    retrieved = repository.find_by_id("non-existent")
    assert retrieved is None


def test_fake_payment_gateway():
    """Test fake payment gateway."""
    gateway = FakePaymentGateway()
    amount = Money(10000, "USD")
    
    result = gateway.process_payment(amount, "order-1")
    
    assert result.success is True
    assert result.transaction_id is not None
    assert result.transaction_id.startswith("txn-")

File: examples/cpp/minimal-order-service/tests/test_adapters.cpp
#include <cassert>
#include <iostream>
#include <memory>
#include "domain/money.hpp"
#include "domain/order.hpp"
#include "adapters/outbound/in_memory_repository.hpp"
#include "adapters/outbound/fake_payment_gateway.hpp"

void test_in_memory_repository() {
    adapters::outbound::InMemoryOrderRepository repository;
    auto total = std::make_shared<domain::Money>(10000, "USD");
    auto order = std::make_shared<domain::Order>("order-1", total);
    
    repository.save(order);
    
    auto retrieved = repository.findById("order-1");
    assert(retrieved != nullptr);
    assert(retrieved->id() == order->id());
    
    std::cout << "✓ test_in_memory_repository passed\n";
}

void test_fake_payment_gateway() {
    adapters::outbound::FakePaymentGateway gateway;
    auto amount = std::make_shared<domain::Money>(10000, "USD");
    
    auto result = gateway.processPayment(amount, "order-1");
    
    assert(result.success);
    assert(!result.transactionId.empty());
    assert(result.transactionId.find("txn-") == 0);
    
    std::cout << "✓ test_fake_payment_gateway passed\n";
}

int main() {
    std::cout << "Running adapter tests...\n";
    test_in_memory_repository();
    test_fake_payment_gateway();
    std::cout << "All adapter tests passed!\n";
    return 0;
}
